.buildkite/cache-builder.yml

```diff
@@ -0,0 +1,36 @@
+# Nodes with values to reuse in the pipeline.
+common_params:
+  # Common plugin settings to use with the `plugins` key.
+  - &common_plugins
+    - automattic/bash-cache#v1.1.0
+    - automattic/git-s3-cache#v1.1.0:
+        bucket: "a8c-repo-mirrors"
+        repo: "wordpress-mobile/wordpress-ios/"
+  # Common environment values to use with the `env` key.
+  - &common_env
+    IMAGE_ID: xcode-12.5.1
+
+steps:
+
+  #################
+  # Build the CocoaPods Base Cache
+  #################
+  - label: ":cocoapods: Rebuild CocoaPods cache"
+    command: |
+      echo "--- :rubygems: Setting up Gems"
+      install_gems
+
+      echo "--- :cocoapods: Rebuilding Pod Cache"
+      cache_cocoapods
+    <<: *common_env
+    <<: *common_plugins
```

Comment:
The `common_env` anchor points to a dictionary that is `IMAGE_ID: xcode-12.5.1`. That dictionary doesn't contain the `env:` key itself, so you need to use the `env:` key at call site here, and use `&common_env` as the content of that `env:` key.

Otherwise, if you use `<<:` like you did here, the `IMAGE_ID: xcode-12.5.1` dictionary will be **merged** with the current dictionary in that context (the one that has `label` and `command` keys and you will end up with a dictionary containing `label`, `command`, and `IMAGE_ID` at the same level.

Similarly for `common_plugins`, I'd actually be surprised is the YAML parser would be happy with that one given that the `&common_plugins` anchor contains not a dictionary, but an array (of 2 items, one being the string `automattic.bash-cache#v1.1.0` and one being a sub-dictionary with a single key `automattic/git-s3-cache#v1.1.0` and a `{bucket:…, repo:…}` as its value). So here with `<<:` you're trying to merge into the current dictionary (the one containing `label` and `command` keys)… an array. You can't merge a dictionary with an array.

PS: And if YAML ends up considering this valid and not erroring on that, it's probably because it does something other than what you want, like maybe in those occasions it falls back to interpreting this `<<:` as an actual key of name `<<` rather than the merge operator since the merge operator wouldn't make sense between a dict and an array)

Reply:
Fixed in 91a05f423af0693ccf1a03f6b8447038eb3bbfc0